src/cpp/src/visual_language/inputs_embedder.cpp

```diff
@@ -284,6 +280,7 @@ class InputsEmbedderMiniCPM : public InputsEmbedder::IInputsEmbedder {
 
         std::vector<ov::Tensor> single_images = to_single_image_tensors(images);
 
+        m_prev_image_id = m_image_id;
```

Comment:
I would store `m_prev_image_id` in `update_chat_history`, as it's responsible for updating of "state"

The same for phi3 changes